### PR TITLE
chore: update subxt and use NodleConfig for txs

### PR DESCRIPTION
**Pull request by aliXsed**
Cover subxt new breaking changes and Ensure we use NodleConfig when composing transactions

### CODE REVIEW

File: src/nodle.rs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    /// Provide a tip to the block author in the chain's native token.
+    #[allow(dead_code)]
+    pub fn tip(mut self, tip: u128) -> Self {
+        self.tip = tip;
+        self
+    }
+
+    /// Allow nonce to be set explicitly.
+    #[allow(dead_code)]
+    pub fn nonce(mut self, nonce: u64) -> Self {
```
Comment (simonsso) [inline src/nodle.rs:134]: There should be a call for this in recent subxt which will take pending extrinsics into account

File: src/nodle.rs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        self
+    }
+
+    /// Build the extrinsic parameters.
+    pub fn build(self) -> <NodleExtrinsicParams<T> as ExtrinsicParams<T>>::Params {
+        let check_mortality_params = if let Some(mortality) = self.mortality {
+            signed_extensions::CheckMortalityParams::mortal(
+                mortality.period,
+                mortality.checkpoint_number,
+                mortality.checkpoint_hash,
```
Comment (simonsso) [inline src/nodle.rs:145]: Is this not limited to a power of two... not sure

File: Cargo.toml [hunk 3]
```diff
@@ -1,18 +1,19 @@
 [package]
```
Comment (simonsso) [inline Cargo.toml:1]: I think we should have a 2 -level cargo.toml with workspace dependencies

**Review by simonsso — APPROVED** [review]

File: src/nodle.rs (hunk 1 above)
Comment (aliXsed) [reply src/nodle.rs:134]: This is only here for setting `nonce` not getting it

File: Cargo.toml (hunk 3 above)
Comment (aliXsed) [reply Cargo.toml:1]: This repo is currently a single package

File: src/nodle.rs (hunk 2 above)
Comment (aliXsed) [reply src/nodle.rs:145]: This is taken from a subxt example.

**Pull request merged**